app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -8,32 +8,62 @@ def create_in_provider(manager_id, params)
         connection.api.credentials.create!(params)
       end
 
+      refresh(manager)
+      find_by!(:resource_id => manager.id, :manager_ref => credential.id)
+    end
+
+    def create_in_provider_queue(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      queue(manager.my_zone, "create_in_provider", [manager_id, params], "Creating #{name}")
+    end
+
+    private
+
+    def refresh(manager)
       # Get the record in our database
       # TODO: This needs to be targeted refresh so it doesn't take too long
       task_ids = EmsRefresh.queue_refresh_task(manager)
       task_ids.each { |tid| MiqTask.wait_for_taskid(tid) }
-
-      find_by!(:resource_id => manager.id, :manager_ref => credential.id)
     end
 
-    def create_in_provider_queue(manager_id, params)
+    def queue(my_zone, method_name, args, action)
```

Comment:
I'd call `my_zone` just `zone` to avoid confusion with the `my_zone` method that exists in some contexts.